### PR TITLE
Refactor processExternalLabels method with slice copy for left labels

### PR DESCRIPTION
**Pull request by songjiayang**

### CODE REVIEW

File: storage/remote/queue_manager.go [hunk 1]
```diff
@@ -629,13 +629,13 @@ func processExternalLabels(ls labels.Labels, externalLabels labels.Labels) label
 			j++
 		}
 	}
-	for ; i < len(ls); i++ {
-		result = append(result, labels.Label{
-			Name:  ls[i].Name,
-			Value: ls[i].Value,
-		})
+
+	if i < len(ls) {
```
Comment (cstyan) [inline storage/remote/queue_manager.go:633]: I think this block could actually just be:

```
result = append(result, ls[i:]...)
result = append(result, externalLabels[j:]...)
```

**Review by cstyan — APPROVED** [review]

File: storage/remote/queue_manager.go (hunk 1 above)
Comment (songjiayang) [reply storage/remote/queue_manager.go]: Thx,  code updated, review please again.

File: storage/remote/queue_manager_test.go [hunk 2]
```diff
@@ -701,6 +701,34 @@ func TestProcessExternalLabels(t *testing.T) {
...
+		},
+
+		// Test empty labels.
+		{
+			labels:         labels.Labels{},
+			externalLabels: labels.Labels{{Name: "a", Value: "b"}},
+			expected:       labels.Labels{{Name: "a", Value: "b"}},
+		},
+
+		// Test labels is longer than externalLabels
```
Comment (bwplotka) [inline storage/remote/queue_manager_test.go:719]:
```suggestion
		// Test labels is longer than externalLabels.
```

File: storage/remote/queue_manager_test.go [hunk 3]
```diff
@@ -701,6 +701,34 @@ func TestProcessExternalLabels(t *testing.T) {
...
+		{
+			labels:         labels.Labels{{Name: "a", Value: "b"}, {Name: "c", Value: "d"}},
+			externalLabels: labels.Labels{{Name: "e", Value: "f"}},
+			expected:       labels.Labels{{Name: "a", Value: "b"}, {Name: "c", Value: "d"}, {Name: "e", Value: "f"}},
+		},
+
+		// Test externalLabels is longer than labels
```
Comment (bwplotka) [inline storage/remote/queue_manager_test.go:726]:
```suggestion
		// Test externalLabels is longer than labels.
```

File: storage/remote/queue_manager.go [hunk 4]
```diff
@@ -629,14 +629,8 @@ func processExternalLabels(ls labels.Labels, externalLabels labels.Labels) label
...
-	for ; i < len(ls); i++ {
-		result = append(result, labels.Label{
-			Name:  ls[i].Name,
-			Value: ls[i].Value,
-		})
-	}
-	result = append(result, externalLabels[j:]...)
-	return result
+	result = append(result, ls[i:]...)
+	return append(result, externalLabels[j:]...)
```
Comment (bwplotka) [inline storage/remote/queue_manager.go:633]: What about simplfying further? 

```suggestion
	return append(append(result, ls[i:]...), externalLabels[j:]...)
```

**Review by bwplotka — APPROVED** [review]
LGTM,  just small nits.

Thanks!

File: storage/remote/queue_manager.go (hunk 4 above)
Comment (songjiayang) [reply storage/remote/queue_manager.go]: nice, so simple.

**Comment by roidelapluie** [comment]
I have closed/reopen for tests to run.

**Comment by songjiayang** [comment]
@roidelapluie ci passed, review again please.

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**